### PR TITLE
[DROOLS-6788] change jbpm Process Instance Id from long to String

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6788

Related PRs
https://github.com/kiegroup/drools/pull/4142
https://github.com/kiegroup/kogito-runtimes/pull/1914

### CODE REVIEW

**Comment by mariofusco** [comment]
The compilation failure in Kogito Apps :: Task Assigning Service is unrelated to this and seems caused by something changed in optaplanner, I already reported the problem.
`
2022-01-20T19:25:44.7630384Z [ERROR] /home/runner/work/drools/drools/kiegroup_kogito_apps/kogito-apps/task-assigning/task-assigning-service/src/test/java/org/kie/kogito/taskassigning/service/SolverExecutorTest.java:[180,13] org.kie.kogito.taskassigning.service.SolverExecutorTest.SolverMock is not abstract and does not override abstract method isEveryProblemChangeProcessed() in org.optaplanner.core.api.solver.Solver`

**Comment by mariofusco** [comment]
Compilation failure caused by optaplanner will be fixed by https://github.com/kiegroup/kogito-apps/pull/1197

File: drools-core/src/main/java/org/drools/core/command/runtime/process/SignalEventCommand.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ public void setEvent(Object event) {
     public Void execute(Context context) {
         KieSession ksession = ((RegistryContext) context).lookup( KieSession.class );
         
-        if (processInstanceId == -1 && correlationKey == null) {
+        if (processInstanceId.equals("-1") && correlationKey == null) {
```
Comment (danielezonca) [inline drools-core/src/main/java/org/drools/core/command/runtime/process/SignalEventCommand.java]:
```suggestion
        if ("-1".equals(processInstanceId) && correlationKey == null) {
```

File: drools-core/src/main/java/org/drools/core/command/runtime/process/SignalEventCommand.java [hunk 2]
```diff
@@ -127,7 +127,7 @@ public Void execute(Context context) {
     }
 
     public String toString() {
-        if (processInstanceId == -1 && correlationKey == null) {
+        if (processInstanceId.equals("-1") && correlationKey == null) {
```
Comment (danielezonca) [inline drools-core/src/main/java/org/drools/core/command/runtime/process/SignalEventCommand.java]:
```suggestion
        if ("-1".equals(processInstanceId) && correlationKey == null) {
```

File: drools-xml-support/src/main/java/org/drools/xml/support/XStreamJSon.java [hunk 3]
```diff
@@ -1303,13 +1303,13 @@ public void marshal(Object object,
                             HierarchicalStreamWriter writer,
                             MarshallingContext context) {
             SignalEventCommand cmd = (SignalEventCommand) object;
-            long processInstanceId = cmd.getProcessInstanceId();
+            String processInstanceId = cmd.getProcessInstanceId();
             String eventType = cmd.getEventType();
             Object event = cmd.getEvent();
 
-            if ( processInstanceId != -1 ) {
+            if ( !processInstanceId.equals("-1") ) {
```
Comment (danielezonca) [inline drools-xml-support/src/main/java/org/drools/xml/support/XStreamJSon.java]:
```suggestion
            if ( !"-1".equals(processInstanceId) ) {
```

File: drools-xml-support/src/main/java/org/drools/xml/support/XStreamXML.java [hunk 4]
```diff
@@ -758,13 +758,12 @@ public void marshal(Object object,
...
             SignalEventCommand cmd = (SignalEventCommand) object;
-            long processInstanceId = cmd.getProcessInstanceId();
+            String processInstanceId = cmd.getProcessInstanceId();
             String eventType = cmd.getEventType();
             Object event = cmd.getEvent();
 
-            if ( processInstanceId != -1 ) {
-                writer.addAttribute( "process-instance-id",
-                                     Long.toString( processInstanceId ) );
+            if ( processInstanceId.equals("-1") ) {
```
Comment (danielezonca) [inline drools-xml-support/src/main/java/org/drools/xml/support/XStreamXML.java]:
```suggestion
            if ( !"-1".equals(processInstanceId) ) {
```

File: kie-api/src/main/java/org/kie/api/runtime/KieRuntime.java [hunk 5]
```diff
@@ -64,4 +64,8 @@ void registerChannel(String name,
     Map< String, Channel> getChannels();
 
     KieSessionConfiguration getSessionConfiguration();
+
+    default ProcessRuntime getProcessRuntime() {
```
Comment (danielezonca) [inline kie-api/src/main/java/org/kie/api/runtime/KieRuntime.java]: Please add comment because it is only used by some test

**Review by danielezonca — APPROVED** [review]

**Pull request merged**